crates/integrations/datafusion/src/table.rs

```diff
@@ -56,6 +56,13 @@ impl IcebergTableProvider {
 
         Ok(IcebergTableProvider { table, schema })
     }
+
+    /// Asynchronously tries to construct a new [`IcebergTableProvider`]
+    /// using the given table. Can be used to create a table provider from an existing table regardless of the catalog implementation.
+    pub async fn try_new_from_table(table: Table) -> Result<Self> {
```

Comment:
@a-agmon Specifically for usage of this API a UT is needed, I think.

Reply:
Sure. Will submit a PR that covers this in a few minutes.
Thanks